### PR TITLE
Fix: Avoid globals starting with 'i/'

### PR DESCRIPTION
**Pull request by StraToN**
Fixes https://github.com/godot-escoria/escoria-issues/issues/153

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc/esc_globals_manager.gd [hunk 1]
```diff
@@ -92,6 +92,15 @@ func set_global(key: String, value, ignore_reserved: bool = false) -> void:
 				"Global key %s is reserved and can not be overridden" % key
 			]
 		)
+	
+	if key.count("/") >= 2 and key.substr(1, 1) == "/":
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc/esc_globals_manager.gd:96]:
```suggestion
	if key.count("/") > 1 and key.begins_with("/"):
```

**Review by BHSDuncan — CHANGES_REQUESTED** [review]
More of me being pedantic. :)

File: addons/escoria-core/game/core-scripts/esc/esc_globals_manager.gd (hunk 1 above)
Comment (StraToN) [reply addons/escoria-core/game/core-scripts/esc/esc_globals_manager.gd:96]: You suggestion is incorrect: the "/" character can only be located at second position in the string, not as the first character.

**Comment by dploeger** [comment]
@StraToN  I'm confused. Wouldn't it make more sense to check in `inventory_add` whether the given global_Id doesn't have a "/" in its name?

**Comment by StraToN** [comment]
@dploeger That would also make sense but in this PR I wanted above all to forbid a global having 2 / characters in its name. Since inventory_add calls set_global, I intended to catch all cases there.

**Comment by dploeger** [comment]
Shouldn't have a global **no** / in the name at all? Even i/ is only a hack to work with inventory items. 🤔

**Comment by StraToN** [comment]
We could indeed modify the global manager to separate globals from inventory globals instead. Technically, nothing blocks the use of / characters in globals. And I'm just thinking that could actually be useful to keep, so that gamedevs can kind of "classify" their globals.

Which would advocate for a change in inventory_add instead as you mentioned. 

I'll change that tomorrow morning.

**Review by BHSDuncan — APPROVED** [review]

**Review by balloonpopper — APPROVED** [review]

**Pull request merged**